functions/index.ts

```diff
@@ -1,87 +1,83 @@
-      callback(null, await DeleteGameHttpResponse(deleteGameData));
+      let deleteGameData = deserializeGameData(userID, JSON.parse(event.body));
+      callback(null, await deleteGameHttpResponse(deleteGameData));
       break;
     default:
-      callback(null, HttpResponse({statusCode: 400, body: JSON.stringify("Invalid operation.")}));
+      callback(null, httpResponse({statusCode: 400, body: JSON.stringify("Invalid operation.")}));
       break;
     }
 }  
 
-export async function GetGameHttpResponse(game: Game) {
-  let response = await GetGame(game);
-  if (response.code) {
-    return ParseDynamoError(response.code);
-  } else if (response.Item) {
-    return HttpResponse({statusCode: 200, body: JSON.stringify(response)});
-  } else {
-    return HttpResponse({statusCode: 404, body: "Unable to get game."});
+export async function getGameHttpResponse(game: Game) {
+  try {
+    let response = await getGame(game);
+    return httpResponse({statusCode: 200, body: JSON.stringify(response)});
+  } catch (err: any) {
+    return httpResponse({statusCode: err.statusCode, body: err.message});
   }
 }
 
-export async function ListGamesHttpResponse() {
-  let response = await ListGames();
-  if (response.code) {
-    return ParseDynamoError(response.code);
-  } else if (response.Items) {
-    return HttpResponse({statusCode: 200, body: JSON.stringify(response)});
-  } else {
-    return HttpResponse({statusCode: 404, body: "Unable to get list of games."})
+export async function listGamesHttpResponse(userID: string) {
+  try {
+    let response = await listGames(userID);
+    return httpResponse({statusCode: 200, body: JSON.stringify(response)});
+  } catch (err: any) {
+    return httpResponse({statusCode: err.statusCode, body: err.message});
   }
 }
 
-export async function CreateGameHttpResponse(game: Game) {
-  let response = await game.CreateGame();
-  if (response.code) {
-    return ParseDynamoError(response.code);
-  } else if (response.Item) {
-    return HttpResponse({statusCode: 201, body: JSON.stringify(response)});
-  } else {
-    return HttpResponse({statusCode: 404, body: "Unable to create game."});
+export async function createGameHttpResponse(game: Game) {
+  try {
+    let response = await game.createGame();
+    return httpResponse({statusCode: 200, body: JSON.stringify(response)});
+  } catch (err: any) {
+    return httpResponse({statusCode: err.statusCode, body: err.message});
   }
 }
 
-export async function ModifyGameHttpResponse(game: Game) {
-  let response = await ModifyGame(game);
-  if (response.code) {
-    return ParseDynamoError(response.code);
-  } else if (response.Attributes) {
-    return HttpResponse({statusCode: 200, body: JSON.stringify(response.Attributes)});
+export async function modifyGameHttpResponse(game: Game) {
+  try {
+    let response = await modifyGame(game);
+    return httpResponse({statusCode: 200, body: JSON.stringify(response)});
```

Comment:
might be nice to return the updated game object in modify (just like create)

Reply:
I thought that's what I did.  The modifyGame() function calls getGame() and returns the game object from that after it updates.